tests/ui/async-await/track-caller/async-closure-gate.rs

```diff
@@ -1,9 +1,27 @@
 // edition:2021
+// revisions: afn nofeat
 
 #![feature(async_closure, stmt_expr_attributes)]
+#![cfg_attr(afn, feature(async_fn_track_caller))]
 
 fn main() {
     let _ = #[track_caller] async || {
         //~^ ERROR `#[track_caller]` on closures is currently unstable [E0658]
     };
 }
+
+#[track_caller]
+async fn foo() {
+    let _ = #[track_caller] async || {
+        //~^ ERROR `#[track_caller]` on closures is currently unstable [E0658]
+    };
+}
+
+#[track_caller]
+async fn foo2() {
+    let _ = async || {
+        let _ = #[track_caller] async || {
```

Comment:
These examples are kinda loaded. I was expecting something that tests things one at a time:
* async fn with a *regular* closure with track caller on it
* regular fn, with an async block, and inside that async block a *regular* closure with track caller on it.
* etc.

The problem with really overloaded tests like this is that they may be failing due to a confluence of tests. Better to separate all the variables and make sure things don't slip thru the cracks.